azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py

```diff
@@ -109,8 +139,20 @@ def _parse_simple_attributes(self, vm_settings):
         self._activity_id = self._string_to_id(vm_settings.get("activityId"))
         self._correlation_id = self._string_to_id(vm_settings.get("correlationId"))
         self._created_on_timestamp = self._ticks_to_utc_timestamp(vm_settings.get("extensionsLastModifiedTickCount"))
+        self._source = vm_settings.get("extensionGoalStatesSource")
+        if self._source is None:
+            self._source = "UNKNOWN"
```

Comment:
Shouldn't this default to Fabric?

Reply:
why? :)

Reply:
(we get this from the HostGAPlugin and do not know the source - we can't assume fabric)

Reply:
haha fair enough :)